templates/tfengine/components/resources/healthcare_datasets/main.tf

```diff
@@ -23,6 +23,37 @@ module "{{resourceName . "name"}}" {
 
   {{hclField . "iam_members" -}}
 
+  {{if has . "consent_stores" -}}
+  consent_stores = [
+    {{range .consent_stores -}}
+    {
+      name = "{{.name}}"
+      {{hclField . "iam_members" -}}
+
+      {{if has . "enable_consent_create_on_update" -}}
+      enable_consent_create_on_update = {
+        {{hcl .enable_consent_create_on_update}}
+      }
+      {{end -}}
+
+      {{if has . "default_consent_ttl" -}}
+      default_consent_ttl = {
+        {{hcl .default_consent_ttl}}
+      }
+      {{end -}}
+
+      {{if $labels := merge (get $ "labels") (get . "labels") -}}
+      labels = {
+        {{range $k, $v := $labels -}}
```

Comment:
Could you try using the `hcl` function here too? You should be able to do `{{hcl $labels}}`